clusterman/aws/auto_scaling_resource_group.py

```diff
@@ -174,16 +148,41 @@ def modify_target_capacity(
 
         autoscaling.set_desired_capacity(**kwargs)
 
-    @property
-    def min_capacity(self) -> int:
-        return self._group_config['MinSize']
+    def scale_up_options(self) -> Iterator[ClustermanResources]:
+        pass
 
-    @property
-    def max_capacity(self) -> int:
-        return self._group_config['MaxSize']
+    def scale_down_options(self) -> Iterator[ClustermanResources]:
+        """ Generate each of the options for scaling down this resource group, i.e. the list of instance types currently
+        running in this resource group.
+        """
+        raise NotImplementedError()
 
-    @timed_cached_property(ttl=CACHE_TTL_SECONDS)
-    def stale_instance_ids(self):
+    def _get_auto_scaling_group_config(self) -> AutoScalingGroupConfig:
+        response = autoscaling.describe_auto_scaling_groups(
+            AutoScalingGroupNames=[self.group_id],
+        )
+        return response['AutoScalingGroups'][0]
+
+    def _get_launch_template_and_overrides(self) -> Tuple[LaunchTemplateConfig, List[InstanceOverrideConfig]]:
+        try:
+            template = self._group_config['LaunchTemplate']
+            launch_template_name = template['LaunchTemplateName']
+            launch_template_version = template['Version']
+            overrides: List[InstanceOverrideConfig] = []
+        except KeyError:
+            policy = self._group_config['MixedInstancesPolicy']
+            template = policy['LaunchTemplate']['LaunchTemplateSpecification']
+            launch_template_name = template['LaunchTemplateName']
+            launch_template_version = template['Version']
+            overrides = policy['Overrides']
+
```

Comment:
Nit: i think the `try` is kind of overbroad here.  It looks like you are only expecting a KeyError on getting `LaunchTemplate`, so maybe you should just put that in the try/except?

Reply:
fixed.